### ISSUE TITLE
Customer grid fails on duplicates

### ISSUE THREAD

**Issue by evs-xsarus** [Datatrics/magento2-integration#22]
**Describe the bug**
Should table datatrics_profile have more records for the same customer ID, it's not possible to load the customer grid in the Magento admin.

**Used versions**
- *Magento Version number(eg 2.3.5)*: 2.4.2
- *Open source/Commerce/B2B*:Commerce
- *Module version number (Check configuration)*: 1.4.0

**To Reproduce**
1. Make sure a customer is synced twice with Datatrics: example:
datatrics_profile
---
| customer_id | store_id | profile_id | address_id | firstname | prefix | lastname | name | company | address | country | city | zip | phone | region | street | created_at | updated_at | update_msg | update_attempts | status | email | 
| ---: | ---: | --- | ---: | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | ---: | --- | --- | 
| 307161 | 1 | cccc:255997:3_32_2_67108864 | 0 |  |  |  |  |  |  |  |  |  |  |  |  | 2021-06-22 13:22:38 | 2021-07-27 11:44:18 |  | 0 | Synced |  | 
| 307161 | 1 | ccc:255997:2 | 0 |  |  |  |  |  |  |  |  |  |  |  |  | 2021-05-01 02:11:09 | 2021-05-01 02:11:09 |  | 0 | Synced |  | 

2. Open the Magento admin
3. Go to the customer overview
> Item (Magento\Customer\Ui\Component\DataProvider\Document) with the same ID "307161" already exists.

**Expected behavior**
The customers should be shown.

**Actual behavior**
The error is shown.

**Screenshots**
n-a

**Additional context**
n-a

**Comment by evs-xsarus** [comment]
Received an update which fixes it:

```diff
--- datatrics/magento2-integration/Plugin/CustomerGridCollection.php
+++ datatrics/magento2-integration/Plugin/CustomerGridCollection.php
@@ -50,6 +50,7 @@
             $leftJoinTableName = $this->resourceConnection->getTableName(ProfileResource::ENTITY_TABLE);
             $collection
                 ->getSelect()
+                ->distinct()
                 ->joinLeft(
                     ['datatrics_profile' => $leftJoinTableName],
                     "datatrics_profile.customer_id = main_table.entity_id",
```

**Comment by Frank-Magmodules** [comment]
Thank you for testing and confirming that our fix works for you @evs-xsarus , we will implement this into the next release!

**Comment by Frank-Magmodules** [comment]
Hi @evs-xsarus ,

We are happy to share that we've just released the new [1.4.1](https://github.com/Datatrics/magento2-integration/releases/tag/v1.4.1) version with a fix for your issue. Thank you for the detailed issue report and proposal for the solution. We are closing this issue now but please feel free to reopen the issue if this still occurs.
